src/allmydata/cli/grid_manager.py

```diff
@@ -0,0 +1,207 @@
+        _config_path_from_option(ctx.parent.params["config"]),
+        ctx.obj.grid_manager,
+    )
+    return 0
+
+
+@grid_manager.command()
+@click.argument("name")
+@click.pass_context
+def remove(ctx, name):
+    """
+    Remove an existing storage-server by name from a Grid Manager
+    """
+    fp = _config_path_from_option(ctx.parent.params["config"])
+    try:
+        ctx.obj.grid_manager.remove_storage_server(name)
+    except KeyError:
+        raise click.ClickException(
+            "No storage-server called '{}' exists".format(name)
+        )
+    cert_count = 0
+    if fp is not None:
+        while fp.child('{}.cert.{}'.format(name, cert_count)).exists():
+            fp.child('{}.cert.{}'.format(name, cert_count)).remove()
+            cert_count += 1
+
+    save_grid_manager(fp, ctx.obj.grid_manager)
+
+
+@grid_manager.command()
+@click.pass_context
+def list(ctx):
+    """
+    List all storage-servers known to a Grid Manager
+    """
+    fp = _config_path_from_option(ctx.parent.params["config"])
+    for name in sorted(ctx.obj.grid_manager.storage_servers.keys()):
+        blank_name = " " * len(name)
+        click.echo("{}: {}".format(name, ctx.obj.grid_manager.storage_servers[name].public_key()))
+        if fp:
+            cert_count = 0
+            while fp.child('{}.cert.{}'.format(name, cert_count)).exists():
+                container = json.load(fp.child('{}.cert.{}'.format(name, cert_count)).open('r'))
+                cert_data = json.loads(container['certificate'])
+                expires = datetime.utcfromtimestamp(cert_data['expires'])
+                delta = datetime.utcnow() - expires
+                click.echo("{}  cert {}: ".format(blank_name, cert_count), nl=False)
+                if delta.total_seconds() < 0:
+                    click.echo("valid until {} ({})".format(expires, abbreviate_time(delta)))
+                else:
+                    click.echo("expired {} ({})".format(expires, abbreviate_time(delta)))
+                cert_count += 1
+
+
+@grid_manager.command()
+@click.argument("name")
+@click.argument(
+    "expiry_days",
+    type=click.IntRange(1, 5*365),  # XXX is 5 years a good maximum?
+)
+@click.pass_context
+def sign(ctx, name, expiry_days):
+    """
+    sign a new certificate
+    """
+    fp = _config_path_from_option(ctx.parent.params["config"])
+    expiry_seconds = int(expiry_days) * 86400
+
+    try:
+        certificate = ctx.obj.grid_manager.sign(name, expiry_seconds)
+    except KeyError:
+        raise click.ClickException(
+            "No storage-server called '{}' exists".format(name)
+        )
+
+    certificate_data = json.dumps(certificate, indent=4)
+    click.echo(certificate_data)
+    if fp is not None:
+        next_serial = 0
+        while fp.child("{}.cert.{}".format(name, next_serial)).exists():
+            next_serial += 1
+        with fp.child('{}.cert.{}'.format(name, next_serial)).open('w') as f:
+            f.write(certificate_data)
```

Comment:
This is race-y.  Python 3 supports the "x" mode to get at the underlying `O_EXCL|O_CREAT` functionality (create the file and open it for writing, or fail if it exists already) but Python 2 does not.  Fortunately `FilePath.create` exists so that's probably the way to go.